### PR TITLE
added basic float and int vector type structs and math module

### PR DESCRIPTION
**Pull request by d-jakob**
Closing #7

### CODE REVIEW

**Comment by d-jakob** [comment]
how do we want functions returning a reference formatted? I prefer grouping the & with the type in this case

File: modules/math/include/m_vec.hh [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+
+namespace Shadowheart {
+namespace Math {
+
+/**
+ * 1D float vector
+ *
+ * TODO detailed description
+ */
+struct vec1 {
```
Comment (SapphireNick) [inline modules/math/include/m_vec.hh:12]: This really should just be `using vec1 = float`. We dont need to overload operators and make a whole struct for what just is one float. The OpenGL functions do also only expect floats/ints etc. in cases where vec1 are used.

File: modules/math/include/m_vec.hh [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+
+  // TODO operator overloading
+}; /* vec4 */
+
+/**
+ * 1D int ivector
+ *
+ * TODO detailed description
+ */
+struct ivec1 {
```
Comment (SapphireNick) [inline modules/math/include/m_vec.hh:126]: Same as for vec1 only now `using ivec1 = int`

File: modules/math/src/m_vec.cc [hunk 3]
```diff
@@ -0,0 +1,31 @@
+#include "m_vec.hh"
+
+namespace Shadowheart {
+namespace Math {
+
+Shadowheart::Math::vec1::vec1(const float &x) : _v{x} {}
+float& Shadowheart::Math::vec1::operator[](const int &i) { return _v[i]; }
```
Comment (SapphireNick) [inline modules/math/src/m_vec.cc:7]: Should not be needed if comments in m_vec.hh are adressed.

File: modules/math/src/m_vec.cc [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+float& Shadowheart::Math::vec2::operator[](const int &i) { return _v[i]; }
+
+Shadowheart::Math::vec3::vec3(const float &x, const float &y, const float &z) : _v{x, y, z} {}
+float& Shadowheart::Math::vec3::operator[](const int &i) { return _v[i]; }
+
+Shadowheart::Math::vec4::vec4(const float &x, const float &y, const float &z, const float &w) : _v{x, y, z, w} {}
+float& Shadowheart::Math::vec4::operator[](const int &i) { return _v[i]; }
+
+Shadowheart::Math::ivec1::ivec1(const int &x) : _v{x} {}
+int& Shadowheart::Math::ivec1::operator[](const int &i) { return _v[i]; }
```
Comment (SapphireNick) [inline modules/math/src/m_vec.cc:19]: Should not be needed again.

**Review by SapphireNick — CHANGES_REQUESTED** [review]
Too much overhead for what just are aliases for existing, basic data-types.

**Comment by SapphireNick** [comment]
> how do we want functions returning a reference formatted? I prefer grouping the & with the type in this case

Agreed. Will (try) to configure our .clang-format when the time for formatting comes for this pr.

**Comment by SapphireNick** [comment]
> > how do we want functions returning a reference formatted? I prefer grouping the & with the type in this case
> 
> Agreed. Will (try) to configure our .clang-format when the time for formatting comes for this pr.

I don't think this is going to work quite like we want it to work...

**Review by SapphireNick — APPROVED** [review]
LGTM !

**Comment by SapphireNick** [comment]
I'm going to run it through the formatting and merge it.

**Pull request merged**